gcloud-java-dns/src/main/java/com/google/gcloud/spi/DefaultDnsRpc.java

```diff
@@ -162,7 +162,9 @@ public Change getChangeRequest(String zoneName, String changeRequestId, Map<Opti
     } catch (IOException ex) {
       DnsException serviceException = translate(ex);
       if (serviceException.code() == HTTP_NOT_FOUND) {
-        if (serviceException.location().equals("entity.parameters.changeId")) {
+        if ((serviceException.location() != null
+            && serviceException.location().equals("entity.parameters.changeId"))
+            || serviceException.getMessage().contains("parameters.changeId")) {
```

Comment:
Can serviceException.getMessage() be null?  If so, we should check for a null value.  To prevent this if from getting too long, you could condense the first two lines into: 

```java
"entity.parameters.changeId".equals(serviceException.location())
```